scheduler/src/cook/mesos/util.clj

```diff
@@ -357,8 +357,8 @@
 ;; get-jobs-by-user-and-state-and-submit is a bit opaque
 ;; because it is reaching into datomic internals. Here is a quick explanation.
 ;; seek-datoms provides a pointer into the raw datomic indices
-;; that we can then seek through. We set the pointer to look
-;; through the avet index, with attribute :job/state, seek to
+;; that we can then seek through. We set the pointer to look through the
+;; vaet index, with value :job.state/{running,waiting,completed}, seek to
```

Comment:
Isn't this function only called for running and waiting, not completed?

Reply:
It can work for any of them, whether its the right query to use is another question. (Datomic doesn't do query optimization. We have to. Fortunately, the data shapes are uniform enough that we can do it statically and do a fair enough job.)

Reply:
What I meant to point out is that the comment states:

> We set the pointer to look through the vaet index, with value :job.state/{running,waiting,completed}

But we don't actually use the value `completed` with this function.

Reply:
Ok. Changed.